src/value-editor/editors/text/directives/text-validations.directive.ts

```diff
@@ -0,0 +1,31 @@
+import { IAttributes, IAugmentedJQuery, INgModelController, IScope } from 'angular';
+import { TextValueEditorComponentController} from '../text.value-editor.component';
+
+/**
+ * @ngdoc directive
+ * @name textValidations
+ * @module angularjs-value-editor.text
+ *
+ * @description
+ *
+ */
+export default class TextValidationsDirective {
+    public static readonly directiveName = 'textValidations';
+
+    public restrict = 'A';
+
+    public require = ['ngModel', '^textValueEditor'];
+
+    public link($scope: IScope,
+                $element: IAugmentedJQuery,
+                $attrs: IAttributes,
+                [ngModelController, controller]: [INgModelController, TextValueEditorComponentController]) {
+        ngModelController.$validators.notBlank = notBlankValidationFactory(controller);
+    }
+}
+
+function notBlankValidationFactory(textValueEditorComponentController: TextValueEditorComponentController) {
+    return (modelValue: string): boolean => {
+        return !textValueEditorComponentController.validations?.notBlank || (modelValue === '' || modelValue?.trim() !== '');
```

Comment:
Tady pozor, je to validace na blank, ne na required, tedy zde vypustit `(modelValue === ''`

Reply:
prave proto tam povoluje prazdny string,
protoze prazdny string po trimu je taky prazdny string a to by potom davalo nevalidni i pro nepovinne pole